### PR TITLE
Remote metadata fetcher

### PR DESCRIPTION
**Pull request by Leechael**

### CODE REVIEW

File: src/features/phat-contract/components/fat-contract-upload-form.tsx [hunk 1]
```diff
@@ -554,7 +557,9 @@ function useUploadCode() {
       setBlueprintPromise(uploadResult.blueprint)
     } catch (err) {
       // TODO: better error handling?
-      if ((err as Error).message.indexOf('Cancelled') === -1) {
+      if ((err as Error).message.indexOf('You need connected to an endpoint & pick a account first.') > -1) {
+        showAccountSelectModal()
```
Comment (Leechael) [inline src/features/phat-contract/components/fat-contract-upload-form.tsx:569]: The `showAccountSelectModal` not yet included in the dependencies array of `useCallback`

File: src/features/phat-contract/components/fat-contract-upload-form.tsx [hunk 2]
```diff
@@ -636,6 +641,18 @@ const Td = tw(ChakraTd)`py-4`
...
+  const handleSign = async () => {
+    try {
+      await requestSign()
+    } catch (err) {
+      if ((err as Error).message.indexOf('You need connected to an endpoint & pick a account first.') > -1) {
+        showAccountSelectModal()
+      }
+    }
+  }
+
```
Comment (Leechael) [inline src/features/phat-contract/components/fat-contract-upload-form.tsx]: Better move to the `useRequestSign` hook internal?

Comment (Leechael) [reply src/features/phat-contract/components/fat-contract-upload-form.tsx]: The connection state and account selection state can be detected before an error is thrown.

Comment (pacoyang) [reply src/features/phat-contract/components/fat-contract-upload-form.tsx]: Not sure the ui logic `useShowAccountSelectModal ` should be write in `useRequestSign` hook internal

File: src/features/phat-contract/argumentsFormAtom.ts [hunk 3]
```diff
@@ -570,6 +569,15 @@ export const getCheckedFieldData = (fieldData: FieldDataNormalized, isRendered:
...
 
+        if (checkedErrors.length && !R.equals(errors, checkedErrors)) {
+          return R.assoc('errors', checkedErrors, fieldData)
+        }
+      } else if (info === TypeDefInfo.Plain) {
+        let checkedErrors = validateNotUndefined(value)
+        if (!checkedErrors.length) {
+          const value$1 = value as string
+          checkedErrors = validatePlainType(fieldData.typeDef, value).errors
+        }
```
Comment (Leechael) [inline src/features/phat-contract/argumentsFormAtom.ts:580]: What this change intent for? It introduce a new bug for the UI.

Comment (pacoyang) [reply src/features/phat-contract/argumentsFormAtom.ts]: revert.

**Pull request merged**